samples/scala-protobuf-tracing/src/main/scala/com/example/ControllerAction.scala

```diff
@@ -0,0 +1,60 @@
+package com.example
+
+import com.example.domain.Post
+import com.google.protobuf.empty.Empty
+import io.opentelemetry.api.trace.StatusCode
+import kalix.scalasdk.action.{Action, ActionCreationContext}
+import org.json4s.native.JsonMethods
+import org.json4s.{DefaultFormats, Formats}
+import sttp.client4._
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+class ControllerAction(creationContext: ActionCreationContext) extends AbstractControllerAction {
+
+  implicit val formats: Formats = DefaultFormats
+
+  val url = "https://jsonplaceholder.typicode.com/posts"
+
+  // tag::create-close-span[]
+  override def callAsyncEndpoint(empty: Empty): Action.Effect[MessageResponse] = {
+    // tag::get-tracer[]
+    val tracerOpt = actionContext.getOpenTelemetryTracer
+    // end::get-tracer[]
+    val span = tracerOpt.map(_.spanBuilder(s"$url/{}")
+      .setParent(actionContext.metadata.traceContext.asOpenTelemetryContext)// <1>
+      .startSpan()// <2>
+      .setAttribute("post", "1"))// <3>
+
+      val responseBody: Future[MessageResponse] = callAsync()
+      responseBody.onComplete {
+        case Failure(exception) =>
+          span.map(_
+            .setStatus(StatusCode.ERROR, exception.getMessage)// <4>
+            .end())// <5>
+        case Success(response) =>
+          span.map(_
+            .setAttribute("result", response.message)// <3>
+            .end())// <5>
+      }
+     effects.asyncReply(responseBody)
+  }
+  // end::create-close-span[]
+
+
+  private def callAsync(): Future[MessageResponse] = {
+    val request = basicRequest.get(uri"${url}/1")
+    val response: Future[Response[Either[String, String]]] = request.send(Main.backend)
```

Comment:
is `Future[Response[Either[String, String]]]` being define by sttp?

Reply:
yeah, they have an  `implicit class sttp.client4.RichRequestTEither` which I imported

Reply:
I ask because it feels super verbose. Shouldn't it be only `Future[Response[String]]`? 

I assume that `Response` is already a type that can express success or failure. Having it hold an `Either` feels overkill. But I might be wrong. I'm not familiar with sttp and I'm sure they added this API for a reason. 

On the other hand, if they have some other implicit that could simplify this return type, it will be better for a sample.

Reply:
you are right. I simplified now. using `quickRequest` that produces a `Request[String]` which in turn produces a `Future[Response[String]]` as you hinted. That is,

```
 val request = quickRequest.get(uri"${url}/1")
    val response = request.send(Main.backend)
    response.map { response =>
      val post = JsonMethods.parse(response.body).extract[Post]
      MessageResponse(post.title)
    }
```